mmcv/image/photometric.py

```diff
@@ -91,3 +91,48 @@ def posterize(img, bits):
     shift = 8 - bits
     img = np.left_shift(np.right_shift(img, shift), shift)
     return img
+
+
+def equalize(img):
+    """Equalize the image histogram.
+
+    This function applies a non-linear mapping to the input image,
+    in order to create a uniform distribution of grayscale values
+    in the output image.
+
+    Args:
+        img (ndarray): Image to be equalized.
+
+    Returns:
+        ndarray: The equalized image.
+    """
+
+    def _scale_channel(im, c, eps=1e-5):
+        """Scale the data in the corresponding channel."""
+        im = im[:, :, c]
+        # Compute the histogram of the image channel.
+        histo = np.histogram(im, 256, (0, 255))[0]
+        # For computing the step, filter out the nonzeros.
+        nonzero_histo = histo[histo > 0]
+        step = (np.sum(nonzero_histo) - nonzero_histo[-1]) // 255
+
+        # Build lut from the full histogram and step.
+        # Compute the cumulative sum, shifted by step // 2
+        # and then normalized by step.
+        lut = (np.cumsum(histo) + (step // 2)) // max(step, eps)
```

Comment:
Consider removing the clip by referring to [here](https://pillow.readthedocs.io/en/stable/_modules/PIL/ImageOps.html#equalize)